java/statestore-commit/src/main/java/sleeper/commit/StateStoreCommitter.java

```diff
@@ -120,4 +124,13 @@ private void apply(IngestAddFilesCommitRequest request) throws StateStoreExcepti
         }
     }
 
+    private void apply(CompactionJobIdAssignmentCommitRequest request) throws StateStoreException {
+        StateStore stateStore = stateStoreProvider.getByTableId(request.getTableId());
+        try {
+            stateStore.assignJobIds(request.getAssignJobIdRequests());
+        } catch (FileReferenceAssignedToJobException e) {
+            LOGGER.info("One or more files are already assigned to a job, skipping request", e);
+            return;
```

Comment:
Looking more closely at the assignJobIds method definition, I think the argument for ignoring this failure also applies for the failure where a file reference no longer exists. Can we test that too?

Reply:
Done, thanks